src/main/kotlin/nl/info/zac/app/planitems/PlanItemsRestService.kt

```diff
@@ -215,57 +219,97 @@ class PlanItemsRestService @Inject constructor(
         val zaak = zrcClientService.readZaak(userEventListenerData.zaakUuid)
         val zaakRechten = policyService.readZaakRechten(zaak)
         assertPolicy(zaakRechten.startenTaak)
-        if (userEventListenerData.restMailGegevens != null) {
+        if (userEventListenerData.restMailGegevens !== null) {
             assertPolicy(zaakRechten.versturenEmail)
         }
         when (userEventListenerData.actie) {
-            UserEventListenerActie.INTAKE_AFRONDEN -> {
-                userEventListenerData.planItemInstanceId?.let {
-                    val planItemInstance = cmmnService.readOpenPlanItem(it)
-                    zaakVariabelenService.setOntvankelijk(planItemInstance, userEventListenerData.zaakOntvankelijk)
-                }
-                if (!userEventListenerData.zaakOntvankelijk) {
-                    zaakService.checkZaakAfsluitbaar(zaak)
-                    val zaakafhandelParameters = zaakafhandelParameterService.readZaakafhandelParameters(
-                        zaak.zaaktype.extractUuid()
-                    )
-                    zgwApiService.createResultaatForZaak(
-                        zaak,
-                        zaakafhandelParameters.nietOntvankelijkResultaattype,
-                        userEventListenerData.resultaatToelichting
-                    )
-                }
-            }
-
-            UserEventListenerActie.ZAAK_AFHANDELEN -> {
-                zaakService.checkZaakAfsluitbaar(zaak)
-                if (!brcClientService.listBesluiten(zaak).isEmpty()) {
-                    val resultaat = zrcClientService.readResultaat(zaak.resultaat)
-                    resultaat.toelichting = userEventListenerData.resultaatToelichting
-                    zrcClientService.updateResultaat(resultaat)
-                } else {
-                    zgwApiService.createResultaatForZaak(
-                        zaak,
-                        userEventListenerData.resultaattypeUuid ?: throw InputValidationFailedException(
-                            errorCode = ErrorCode.ERROR_CODE_VALIDATION_GENERIC,
-                            message = "Resultaattype UUID moet gevuld zijn bij het afhandelen van een zaak."
-                        ),
-                        userEventListenerData.resultaatToelichting
-                    )
-                }
-            }
+            UserEventListenerActie.INTAKE_AFRONDEN -> this.handleIntakeAfronden(zaak, userEventListenerData)
+            UserEventListenerActie.ZAAK_AFHANDELEN -> this.handleZaakAfhandelen(zaak, userEventListenerData)
         }
+
         userEventListenerData.planItemInstanceId?.let {
             cmmnService.startUserEventListenerPlanItem(it)
         }
-        if (userEventListenerData.restMailGegevens != null) {
+        if (userEventListenerData.restMailGegevens !== null) {
```

Comment:
If this extra `=` intended? Maybe better to use `userEventListenerData.restMailGegevens?.run { }` here?